spec/models/friendship_spec.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Friendship, type: :model do
+  before :each do
+    @user1 = User.create(name: 'noomdalv', email: 'noom@dalv.com',
+                         password: '123456', password_confirmation: '123456')
+    @user2 = User.create(name: 'suman', email: 'suman@test.com',
+                         password: 'abcdef', password_confirmation: 'abcdef')
+    @friendship = @user1.friendships.build(friend_id: @user2.id)
+  end
+
+  context 'Friendship validation'
+  it 'Valid Friendship Check' do
+    expect(@friendship).to be_valid
+  end
```

Comment:
What about the context for when the friendship is supposed to be invalid? Like when a user tries to send friend request to a user that does not exist or to himself? :wink:  What if a user sends a friend request to a user who is already a friend? :man_shrugging: